jquery.flot.touchNavigate.js

```diff
@@ -14,175 +14,369 @@
     }
 
     function initTouchNavigation(plot, options) {
-        var prevPanX = 0,
-            prevPanY = 0,
-            prevTapX = 0,
-            prevTapY = 0,
-            prevDist = null,
+        var prevDist = null,
             twoTouches = false,
             prevTapTime = 0,
             axis = null,
             prevAxisTouched = 'none',
             xAxisTouched = false,
-            yAxisTouched = false;
+            yAxisTouched = false,
+            prevPan = { x: 0, y: 0 },
+            prevTap = { x: 0, y: 0 };
+
+        function getMidPoint(e) {
```

Comment:
or better after the initTouch function as id preferable to read the high level stuff first, then the low level details of how this is implemented